jgrapht-core/src/main/java/org/jgrapht/experimental/dag/DirectedAcyclicGraph.java

```diff
@@ -532,6 +533,82 @@ private void reorder(Set<V> df, Set<V> db, Visited visited)
     }
 
     /**
+     * This function is implemented recursively. Therefore only graphs without
```

Comment:
(since this is on DAG now, you can delete the part of the comment about no cycles allowed)